ethereum/attestation-test/contracts/VerifyTicket.sol

```diff
@@ -200,18 +193,52 @@ contract VerifyTicket {
 
         (length, decodeIndex, ) = decodeLength(attestation, decodeIndex + length); // Issuer Sequence (14) [[2.5.4.3, ALX]]], (Issuer: CN=ALX)
 
-        (length, decodeIndex, ) = decodeLength(attestation, decodeIndex + length); // Validity time
+        (decodeIndex, timeStampValid) = decodeTimeBlock(attestation, decodeIndex + length, ticketId);
 
-        (length, decodeIndex, ) = decodeLength(attestation, decodeIndex + length); // Smartcontract?
+        (length, decodeIndex, ) = decodeLength(attestation, decodeIndex); // Smartcontract?
 
         (subject, decodeIndex) = addressFromPublicKey(attestation, decodeIndex + length);
 
+        commitment1 = decodeCommitment(attestation, decodeIndex);
+    }
+
+    function decodeCommitment (bytes memory attestation, uint256 decodeIndex) internal virtual pure returns (bytes memory commitment) {
+
+        uint256 length ;
+        
         (length, decodeIndex, ) = decodeLength(attestation, decodeIndex); // ID code (07)
-        (commitment1, decodeIndex) = recoverCommitment(attestation, decodeIndex + length); // Commitment 1, generated by
-                                                                                           // IdentifierAttestation constructor
+        (commitment, ) = recoverCommitment(attestation, decodeIndex + length); // Commitment 1, generated by
+        // IdentifierAttestation constructor
     }
 
-    function recoverCommitment(bytes memory attestation, uint256 decodeIndex) private pure returns(bytes memory commitment, uint256 resultIndex)
+    function decodeTimeBlock(bytes memory attestation, uint256 decodeIndex, bytes memory ticketId) public view returns (uint256 index, bool valid)
+    {
+        bytes memory timeBlock;
+        uint256 length;
+        uint256 blockLength;
+        bytes1 tag;
+
+        (blockLength, index, ) = decodeLength(attestation, decodeIndex); //30 32
+        (length, decodeIndex, ) = decodeLength(attestation, index); //18 0f
+        (length, timeBlock, decodeIndex, tag) = decodeElement(attestation, decodeIndex + length); //INTEGER_TAG if blockchain friendly time is used
+        if (tag == INTEGER_TAG)
+        {
+            uint256 startTime = bytesToUint(timeBlock);
+            (length, decodeIndex, ) = decodeLength(attestation, decodeIndex); //18 0F
+            (, timeBlock, decodeIndex,) = decodeElement(attestation, decodeIndex + length);
+            uint256 endTime = bytesToUint(timeBlock);
+            valid = block.timestamp > startTime && block.timestamp < endTime;
+        }
+        else
```

Comment:
After discussion with @oleggrib we agreed that the best is simply to fail if the attestation does not contain a blockchain friendly timestamp.

Reply:
@jot2re , in this case better make validityInteger as required value, not optional in the Java lib.

Reply:
Good point! Currently we always include it. The optionality is only to reflect that the old attestations technically are correct (and should work in the browser), such that the spec is backward compatible. But I am changing things now, so that it is not possible to set it to false when constructing  new attestations, but the old ones can still be decoded.